icu4c/source/i18n/unum.cpp

```diff
@@ -113,6 +112,22 @@ unum_open(  UNumberFormatStyle    style,
     case UNUM_DURATION:
         retVal = new RuleBasedNumberFormat(URBNF_DURATION, Locale(locale), *status);
         break;
+
+    case UNUM_NUMBERING_SYSTEM: {
+        // if the locale ID specifies a numbering system, go through NumberFormat::createInstance()
+        // to handle it properly (we have to specify UNUM_DEFAULT to get it to handle the numbering
+        // system, but we'll always get a RuleBasedNumberFormat back); otherwise, just go ahead and
+        // create a RuleBasedNumberFormat ourselves
+        UErrorCode err = U_ZERO_ERROR;
+        Locale localeObj(locale);
+        char keywordBuffer[20];
+        int32_t keywordLength = localeObj.getKeywordValue("numbers", keywordBuffer, 20, err);
```

Comment:
optional:
You could get false failures here from over-long keyword values.
In newer code, we generally use a CharStringByteSink and [this version of getKeywordValue()](https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classicu_1_1Locale.html#aef599483560e19d4f0ac8a65e0a4ddd4).

Reply:
Does this work?

```cpp
UErrorCode localStatus = U_ZERO_ERROR;
int32_t keywordLength = localeObj.getKeywordValue("numbers", nullptr, 0, localStatus);
// ignore localStatus: expected to be U_BUFFER_OVERFLOW_ERROR
```

Reply:
I like Shane's option better, since I don't need the actual value, and it seems to work.